doc/design/secret-provisioning.md

```diff
@@ -0,0 +1,167 @@
+# Flexible Secret provisioning with Mystikos
+
+For confidential computing applications, we don't recommend that the
+application code to contain any secrets, as the code (either in source
+or binary forms) could be exposed to hostile environment during
+downloading, uploading, copying, or launching of the application.
+Instead, we recommend the secrets to be securely released to the
+application while it is executed in a Trused Execution Environment (TEE).
+The providers of the secrets could be a key vault service such as AKV,
+M-HSM, or a service run by the user in a locked down and trusted
+environment.
+
+This design document describes the process of secret provisioning
+ with Mystikos in 5 aspects:
+
+1. **the user interface**: how does Mystikos allow users to specify the
+secrets they want in their applications?
+1. **the client library**: how does Mystikos provide a reference SSR
```

Comment:
What is SSR?

Reply:
Secure Secret Release. Now should be replaced with "Secret Provisioning".